### ISSUE TITLE
How to avoid wstxns1: prefix 

### ISSUE THREAD

**Issue by mailsanchu** [FasterXML/jackson-dataformat-xml#666]
Given the following classes 
RootNamespace.java
```
@JacksonXmlRootElement(namespace = "namespace1", localName = "RootNamespace")
public class RootNamespace {
    @JacksonXmlProperty(namespace = "namespace1", localName = "childNamespace")
    private ChildNamespace childNamespace;
}
```

ChildNameSpace.java

```
public class ChildNamespace {
    @JacksonXmlProperty(namespace = "namespace2", localName = "Name")
    private ThirdLevel thirdLevel;
}
```

ThirdLevel.java

```
@Setter
@Getter
public class ThirdLevel {
    @JacksonXmlProperty(namespace = "namespace2", localName = "Name")
    private String name;
}
```

and It is generating the following output when I convert it into string 

```
<?xml version="1.0" encoding="UTF-8"?>
<RootNamespace xmlns="namespace1">
  <childNamespace>
    <wstxns1:Name xmlns:wstxns1="namespace2"/>
  </childNamespace>
</RootNamespace>
```
and I would like to generate it the following 
```
<RootNamespace xmlns="namespace1">
    <childNamespace>
        <Name xmlns="namespace2"/>
    </childNamespace>
</RootNamespace>
```
Is it possible to do. Any help is greatly appreciated.

**Comment by cowtowncoder** [comment]
To do this, you will probably need to construct `XMLStreamWriter`, initialize it with namespace binding, and pass that to `XMLMapper.writeValue()`. I don't think there is a way to avoid automated namespace declaration generation.
There is a way in Woodstox to re-configure prefix to use, I think.

**Comment by norrisjeremy** [comment]
Do the changes in #207 provide any ability to accomplish this somehow?

**Comment by cowtowncoder** [comment]
@norrisjeremy Yes -- you can declare namespace bindings: either "default" namespace:

    new XmlGeneratorInitializer()
                .addDefaultNamespace("urn:produce:fruit"))

or prefixed

    new XmlGeneratorInitializer()
                .addNamespace("fruit", "urn:produce:fruit"));

and then declarations get added using prefix indicated (or none for "default").

**Comment by norrisjeremy** [comment]
I don't think that quite accomplishes what's being asked here though?

Instead of this:
```
<RootNamespace xmlns="namespace1">
  <childNamespace>
    <wstxns1:Name xmlns:wstxns1="namespace2"/>
  </childNamespace>
</RootNamespace>
```

We want to omit the `wstxns1` prefix for `namespace2` and instead emit this:
```
<RootNamespace xmlns="namespace1">
  <childNamespace>
    <Name xmlns="namespace2"/>
  </childNamespace>
</RootNamespace>
```

I wasn't sure if the changes to add `XmlGeneratorInitializer()` would somehow allow coercing the output to omit prefixes entirely for child elements with differing namespaces?

**Comment by cowtowncoder** [comment]
Correct, you can only specify one non-prefixed ("default") namespace using `XmlGeneratorInitializer` mechanism.
There is no support (yet) for "always try to use default namespace binding for [all] elements".

**Comment by norrisjeremy** [comment]
Ok, thanks for confirming!
